core/bin/external_node/src/main.rs

```diff
@@ -208,6 +215,37 @@ async fn init_tasks(
         }
     }));
 
+    let pruning_enabled = config.optional.l1_batch_age_to_prune_hours.is_some();
+    if pruning_enabled {
+        let l1_batch_age_to_prune =
+            Duration::from_secs(3600 * config.optional.l1_batch_age_to_prune_hours.unwrap());
+        let db_pruner = DbPruner::new(
+            DbPrunerConfig {
+                soft_and_hard_pruning_time_delta: Duration::from_secs(60),
+                pruned_chunk_size: config.optional.pruning_chunk_size,
+                next_iterations_delay: Duration::from_secs(5),
+            },
+            vec![
+                Arc::new(L1BatchExistsCondition {
+                    conn: connection_pool.clone(),
+                }),
+                Arc::new(NextL1BatchHasMetadataCondition {
+                    conn: connection_pool.clone(),
+                }),
+                Arc::new(NextL1BatchWasExecutedCondition {
+                    conn: connection_pool.clone(),
+                }),
+                Arc::new(L1BatchOlderThanPruneCondition {
+                    minimal_age: l1_batch_age_to_prune,
+                    conn: connection_pool.clone(),
+                }),
```

Comment:
It would make the component significantly less testable, now we can test the pruning logic without conditions, some solution would be to introduce a new "wrapper" type that would set those conditions by default, but I really like those conditions being a parameter of the pruner rather than something built-in into it, It's nice to be able to immediately see the pruning conditions

Reply:
Public interface just accepts a vector of conditions, I really don't think this interface if complex